pinc/wordcheck_engine.inc

```diff
@@ -378,23 +378,23 @@ function get_bad_words_via_pattern($input_words_w_freq, $languages)
 {
     list($langcode3s, $messages) = get_langcode3s_for_languages($languages);
 
-    $badWords = array();
+    $badWords = [];
 
     // pattern #1
     // flag all words with embedded digits while
     // ignoring common valid <digit><letter> combinations
     //
-    $exceptions_for_lang = array(
+    $exceptions_for_lang = [
         'eng' => '\d*[04-9]th|\d*1[123]th|(\d*[02-9])?(1st|2nd|3rd)',
         'fre' => '\d+(er|me)',
         // etc
-    );
-    $exceptions = array();
+    ];
+    $exceptions = [];
     foreach($langcode3s as $langcode3)
     {
         $exceptions[] = array_get($exceptions_for_lang, $langcode3, "");
     }
-    $exceptions = implode('|', $exceptions);
+    $exceptions = join('|', $exceptions);
```

Comment:
why this change?

Reply:
No great reason. I just trip over `implode` every time I see it. Every other language I know that has a similar feature calls it `join` and the word feels less jargony and more pseudocode to me. I figured that since I was doing the cosmetic `array(...)` to `[...]` conversion I may as well do this too. (And of course, PHP being PHP, `split` had to be a different, now deprecated function that isn't the inverse of `join`, so we're stuck with `explode`. Sigh.)

Reply:
I'm neutral on these changes. I'm happy to take them but I don't feel like we need to search/replace to change them.